### PR TITLE
G3 2024 v2 #14556 Error message for wrong credentials

### PR DESCRIPTION
**Pull request by a22moamy**
# The fix
There should now be an error message stating "Wrong username or password" when the user enters the wrong credentials for a user. This message can be seen below (the red text):
Skärmavbild 2024-04-15 kl  12 44 18

Please test that this message also appears for you (the tester) when you enter the wrong username or password when trying to log in. 

# New issues found
1. While logged out, an error message related to the fetching of github repos will be emitted every second, should this be fixed? The error message:
Skärmavbild 2024-04-15 kl  12 49 15

2. In the code, there seems to be an attempt to display an error message when the user has reached the log in limit. However, I found that I can spam the login button with the wrong credentials without ever getting this message. Perhaps a new issue should be made to fix this. 
The attempt can be found in the `processLogin()` function from `dugga.js` and looks as follows: 
```
		else if(result['login'] == "limit"){
        	displayAlertText("#login #message", "Too many failed attempts, <br /> try again later");
		}
```

3.  Should this text be removed from the login popup since it seems like you can log in from multiple pages and not just the course page? The text:
Skärmavbild 2024-04-15 kl  12 47 58
 Alternatively, there could be an issue investigating which pages you CANNOT log in from, or what pages would cause errors when logging in from?

### CODE REVIEW

**Review by a22danwe — COMMENTED** [review]
Looks good. i like the effects

**Review by a22danwe — APPROVED** [review]

File: Shared/dugga.js [hunk 1]
```diff
@@ -1599,31 +1599,40 @@ function processLogin() {
...
+//Displays an error text and makes login inputs flash red
+function loginFail(){
+	displayAlertText("#login #message", "Wrong username or password");
+	$("input#username").addClass("loginFail");
+	$("input#password").addClass("loginFail");
+	setTimeout(function(){
+	$("input#username").removeClass("loginFail");
+	$("input#password").removeClass("loginFail");
+			}, 2000);
+}
```
Comment (a22willi) [inline Shared/dugga.js:1635]: Fix indentation.

File: Shared/dugga.js [hunk 2]
```diff
@@ -1599,31 +1599,40 @@ function processLogin() {
...
-        console.log("error");
+	  error: function(data) {
+        console.log("Login error status: "+data.status);
+		// status 200 = successful request
+        if(data.status != 200){
+			loginFail();
+        } 
       }
     });
 }
```
Comment (a22willi) [inline Shared/dugga.js:1623]: Fix indentation.

**Review by a22willi — CHANGES_REQUESTED** [review]
Some minor formatting issues.

**Review by a22willi — APPROVED** [review]

**Pull request merged**